File: CTF/2025/QnQsec/secure-letter-revenge/bot.js
const puppeteer = require('puppeteer');
const express = require('express');

const app = express();
app.use(express.json());

const BOT_CONFIG = {
    baseUrl: process.env.WEB_URL || 'http://localhost:3001',
    cookieName: 'flag',
    cookieValue: process.env.FLAG || 'FLAG{test_flag}'
};

// Start the bot server
app.get('/report', async (req, res) => {
    const url = req.query.url;
    
    if (!url) {
        return res.status(400).json({ error: 'URL is required' });
    }

    console.log(`🤖 Bot received report for URL: ${url}`);
    
    try {
        await visitUrl(url);
        res.json({ 
            success: true, 
            message: 'Bot visited the URL successfully'
        });
    } catch (error) {
        console.error('❌ Bot error:', error.message);
        res.status(500).json({ 
            success: false, 
            error: 'Bot failed to visit URL' 
        });
    }
});

async function visitUrl(url) {
    console.log('🚀 Starting bot browser...');
    
    const browser = await puppeteer.launch({
        headless: true,
        args: [
            '--no-sandbox',
            '--disable-setuid-sandbox',
            '--disable-dev-shm-usage',
            '--disable-accelerated-2d-canvas',
            '--no-first-run',
            '--no-zygote',
            '--disable-gpu'
        ]
    });

    try {
        const page = await browser.newPage();
        
        await page.setViewport({ width: 1280, height: 720 });
        
        await page.setCookie({
            name: BOT_CONFIG.cookieName,
            value: BOT_CONFIG.cookieValue,
            domain: process.env.COOKIE_DOMAIN || 'localhost',
            path: '/',
            httpOnly: false,
            secure: false
        });

        console.log(`🍪 Set cookie: ${BOT_CONFIG.cookieName}=${BOT_CONFIG.cookieValue}`);
        console.log(`🎯 Visiting URL: ${url}`);

        await page.goto(url, { 
            waitUntil: 'networkidle2',
            timeout: 10000 
        });

        await new Promise(resolve => setTimeout(resolve, 15000));

        const title = await page.title();
        console.log(`📄 Page title: ${title}`);

        page.on('console', msg => {
            console.log(`🖥️  Page console: ${msg.text()}`);
        });

        page.on('request', request => {
            if (request.url().includes('webhook') || request.url().includes('analytics')) {
                console.log(`🌐 Network request: ${request.url()}`);
            }
        });

        console.log('✅ Bot visit completed successfully');

    } finally {
        await browser.close();
        console.log('🔒 Browser closed');
    }
}

// Health check endpoint
app.get('/health', (req, res) => {
    res.json({ 
        status: 'healthy', 
        bot: 'ready',
    });
});

// Start the bot server
const PORT = process.env.PORT || 3002;
app.listen(PORT, () => {
    console.log(`🤖 XSS Bot started on port ${PORT}`);
    console.log(`🎯 Bot will visit URLs reported to http://localhost:${PORT}/report`);
    console.log(`🍪 Bot cookie: ${BOT_CONFIG.cookieName}=${BOT_CONFIG.cookieValue}`);
});

module.exports = app;

File: CTF/2026/0xL4ugh/0xClinic_-_Player/Player/bot.py
import asyncio
from pyppeteer import launch
import time
from utils.auth import create_token
import sys
from os import environ

stdout = sys.stdout
stderr = sys.stderr

async def run_health_check(DOCTOR_SESSION, ADMIN_KEY):
    browser = await launch(
        headless=True,
        args=[
            '--no-sandbox',
            '--disable-dev-shm-usage',
            '--disable-gpu',
        ],
        ignoreHTTPSErrors=True,
    )
    
    page = await browser.newPage()
    await page.setCookie({
        'name': 'ADMIN_KEY',
        'value': ADMIN_KEY,
        'domain': 'localhost',
        'path': '/',
        'httpOnly': True,
        'secure': False,
        'sameSite': 'Lax'})
    
    await page.setCookie({
        'name': 'auth',
        'value': DOCTOR_SESSION,
        'domain': 'localhost',
        'path': '/',
        'httpOnly': True,
        'secure': False,
        'sameSite': 'Lax'
    })
    await page.setRequestInterception(True)

    async def intercept(request):
        url = request.url
        if (
            url.startswith("http://localhost/")
        ):
            await request.continue_()
        else:
            await request.abort()

    page.on("request", lambda req: asyncio.ensure_future(intercept(req)))

    try:
        response = await page.goto('http://localhost/api/health', {'waitUntil': 'load', 'timeout': 15000})
        
        if response.status == 200:
            print("[*] Health check page loaded successfully.", flush=True)
        else:
            print(f"[!] Failed to load health check page. Status: {response.status}", flush=True)
    
    except Exception as e:
        print(f"[!] Error during navigation: {e}", flush=True)
        print(f"[!] Error type: {type(e).__name__}", flush=True)
        
    await asyncio.sleep(3)
    await browser.close()

async def main():
    print("[*] Generating doctor session token...", flush=True)
    DOCTOR_SESSION = create_token({
        "user_id": "D001", "username": "HealthCheckBot", "role": "doctor", "verified": True},sys.argv[1])

    try:
        ADMIN_KEY = environ["ADMIN_KEY"]
    except KeyError:
        exit("[!] ADMIN_KEY not set in environment variables.")

    print("[*] ADMIN_KEY:", ADMIN_KEY, flush=True)

    print(f"\n[*] Running health check at {time.strftime('%Y-%m-%d %H:%M:%S')}", flush=True)
    try:
        await run_health_check(DOCTOR_SESSION, ADMIN_KEY)
    except Exception as e:
        print(f"[!] Health check failed: {e}", flush=True)

if __name__ == "__main__":
    asyncio.run(main())
    sys.exit(0)